tools/update_fake_backends.py

```diff
@@ -49,17 +53,221 @@ def default(self, o):
 )
 
 
+class _PropertyStats:
+    def __init__(self, properties):
+        self.properties = properties
+        self.no_qubits = len(properties.qubits)
+        self.no_gates = len(properties.gates)
+        self._gate_errors = None  # {gate_name: [qubits]}
+        self._gate_total = None  # {gate_name: total_amount_of_them}
+        self._readout_errors = None  # {qubit}
+
+    @property
+    def name(self):
+        """backend name"""
+        return self.properties.backend_name
+
+    @property
+    def date(self):
+        """backend last update date"""
+        return self.properties.last_update_date
+
+    @property
+    def readout_errors(self):
+        """list[qubits] with the readout==1"""
+        if self._readout_errors is None:
+            self._readout_errors = set()
+            for qubit in range(self.no_qubits):
+                if self.properties.readout_error(qubit) == 1:
+                    self._readout_errors.add(qubit)
+        return self._readout_errors
+
+    @property
+    def total_readout_rate(self):
+        """tuple(witherror/total/percentage"""
+        return (
+            len(self.readout_errors),
+            self.no_qubits,
+            100 * len(self.readout_errors) / self.no_qubits,
+        )
+
+    @property
+    def gate_errors(self):
+        """dict gate->list[qubits] with the gate_error==1 for each gate"""
+        if self._gate_errors is None:
+            self._count_gate_errors()
+        return self._gate_errors
+
+    def gate_rate(self, gate_name):
+        """gate_name -> tuple(with_error, total, percentage)"""
+        return (
+            len(self.gate_errors[gate_name]),
+            self.gate_total[gate_name],
+            100 * len(self.gate_errors[gate_name]) / self.gate_total[gate_name],
+        )
+
+    @property
+    def total_gate_rate(self):
+        """tuple(with_error, total, percentage)"""
+        with_error = sum([len(e) for e in self.gate_errors.values()])
+        return (with_error, self.no_gates, 100 * with_error / self.no_gates)
+
+    @property
+    def gate_total(self):
+        """dict gate->int with the total amount for each gate"""
+        if not self._gate_total:
+            self._count_gate_errors()
+        return self._gate_total
+
+    def _count_gate_errors(self):
+        """calculates:
+        * gate_total
+        * gate_errors"""
+        self._gate_total = defaultdict(lambda: 0)
+        self._gate_errors = defaultdict(lambda: [])
```

Comment:
Just for ease in the future: `int` and `list` respectively already have the behaviour you want and are a bit more descriptive - `int() == 0` and `defaultdict(int)` reads naturally.

Reply:
of course! thanks!